src/js/setUpSite.ts

```diff
@@ -214,26 +247,36 @@ const setUpCitiesLayer = async (map, parkingLayer) => {
 
   // Set up map to update when city selection changes.
   const cityToggleElement = document.getElementById("city-choice");
-  cityToggleElement.addEventListener("change", async () => {
-    const cityId = cityToggleElement.value;
-    snapToCity(map, cities[cityId].layer);
-  });
+  if (cityToggleElement instanceof HTMLSelectElement) {
+    cityToggleElement.addEventListener("change", async () => {
+      const cityId = cityToggleElement.value;
+      const { layer } = cities[cityId];
+      if (layer) {
+        snapToCity(map, layer);
+      }
+    });
 
-  // Set up map to update when user clicks within a city's boundary
-  allBoundaries.addEventListener("click", (e) => {
-    const currentZoom = map.getZoom();
-    if (currentZoom > 7) {
-      const cityId = e.sourceTarget.feature.properties.id;
-      cityToggleElement.value = cityId;
-      snapToCity(map, cities[cityId].layer);
-    }
-  });
+    // Set up map to update when user clicks within a city's boundary
+    allBoundaries.addEventListener("click", (e) => {
+      const currentZoom = map.getZoom();
+      if (currentZoom > 7) {
+        const cityId = e.sourceTarget.feature.properties.id;
+        cityToggleElement.value = cityId;
+        const { layer } = cities[cityId];
+        if (layer) {
+          snapToCity(map, layer);
+        }
+      }
+    });
 
-  // Load initial city.
-  const cityId = cityToggleElement.value;
-  setUpAutoScorecard(map, cities, parkingLayer);
-  snapToCity(map, cities[cityId].layer);
-  setScorecard(cityId, cities[cityId]);
+    // Load initial city.
+    const cityId = cityToggleElement.value;
+    setUpAutoScorecard(map, cities, parkingLayer);
+    snapToCity(map, cities[cityId].layer);
+    setScorecard(cityId, cities[cityId]);
+  } else {
+    throw new Error("#city-choice is not a select element");
+  }
```

Comment:
Nit: to avoid nesting like this, you can do an early return. Invert the conditional to check if this error condition is true and throw an exception. Then you can avoid indenting the above.

Reply:
I'll be honest I'm not sure how crazy I am about early return in this instance.  In this case I feel the indentation makes it clearer that the cityToggleElement has changed type within the scope of the curly braces by narrowing.  That said I'm in favor of resolving discussions about style by favoring consistency with the style of the rest of the organization...so if early return is the preference I'm happy to make that change, I just wanted to state my thought process for the record.